microshift.sh

```diff
@@ -54,12 +54,11 @@ ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
 # podman package is required to run the ostree-container to serve the rpm-ostree content
 # createrepo package is required to create localrepo for microshift and it's dependenices
 # yum-utils package is required for reposync utility to synchronize packages of a remote DNF repository to a local directory
-# containernetworking-plugins contains networking plugin like bridge which required by podman
-# runc package provide OCI spec for running containers
 function configure_host {
     sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
-      cockpit-composer cockpit-machines bash-completion lorax podman \
-      yum-utils createrepo runc containernetworking-plugins
+      cockpit-composer cockpit-machines bash-completion lorax \
+      yum-utils createrepo
+    sudo dnf install -y podman --setopt=install_weak_deps=True
```

Comment:
```
$ grep -nR runc podman.spec
<=== no result ===>
```

from the commit log:

> but podman's spec file recommends crun and not runc, no idea if these are the same or not.

Reply:
but if `runc` is not installed even with weak dependency then bundle creation job on microshift fails yesterday I also ask on internal channel why `runc` required as part of https://github.com/openshift/microshift/commit/daa3cfc959f516c80884cd0f4b33df76edfa540c but I only get it is not around podman but crio command doesn't work without default runtime. Also https://github.com/openshift/microshift/commit/a52e38f9ad1c33d52105b01e94644030e5919c67 is where containernetworking-plugins package is added as part of image-builder dep but I don't see it on official doc.

Reply:
> but if runc is not installed even with weak dependency then bundle creation job on microshift fails

podman intentionally switched from runc to crun, hopefully microshift is going to be fine with both:
```
* Mon May 23 2022 Jindrich Novy <jnovy@redhat.com> - 2:4.1.0-4
- don't require runc and Recommends: crun
```
Or have you tried already with crun instead of runc, and this  fails?

Reply:
The error message you mentioned in the commit log is
```
$ scripts/image-builder/cleanup.sh -full
Error: default OCI runtime "runc" not found: invalid argument
```

which is coming from `podman`, not from `microshift`, and I don't see anything in `cleanup.sh` forcing runc over crun, so hopefully any runtime will be fine.

Reply:
I would still test this on fresh 9.2 system because the error coming from `cleanup.sh` and we don't have control over there. I am still confused why the podman required `runc` when it should use `crun` by default, is it somewhere missed as part of config or some other place?

Reply:
Fwiw, microshift releases notes says: « Switching from runc to crun and cgroupv2 » so `runc` is definitely not the best package to use.

> I am still confused why the podman required runc when it should use crun by default, is it somewhere missed as part of config or some other place?

It is some last resort fallback on systems which are missing required packages, maybe it's there for backwards compat, maybe it's there because nobody noticed, ...

Reply:
> `Error: default OCI runtime "runc" not found: invalid argument`

`/usr/share/containers/containers.conf` from `containers-common-3:1-35.rhaos4.13.el9.x86_64` has: `runtime = "runc"` which is causing this. We should not be using the rhaos one on the host.

Reply:
Latest iteration of my PR should no longer have this problem (assuming the rhaos repos were not enabled before running microshift.sh).